miniapp/miniapp_recipes.cpp

```diff
@@ -82,6 +82,11 @@ class basic_cell_recipe: public recipe {
     cell_size_type num_cells() const override { return ncell_; }
 
     util::unique_any get_cell(cell_gid_type i) const override {
+        // The last 'cell' is a fs_cell with one spike at t=0
+        if (i == ncell_-1) {
```

Comment:
see above comments. Not counting the spike source in `ncell_` removes quite a few of these `-1` expressions.